src/main/java/ims/cs/qsample/run/QSample.java

```diff
@@ -201,6 +242,38 @@ public static void main(String[] args) throws ClassNotFoundException, ParserConf
                 RunPerceptronSampler.runPsPipeline(null, testDocs, null, null,
                         StaticConfig.beginMargin, StaticConfig.endMargin, StaticConfig.cueMargin, perceptrons);
             }
+        } else if (StaticConfig.cliMode == CliMode.CONSOLE) {
+            // load common model
+            QuotationPerceptrons perceptrons = Common.deserializeModels(StaticConfig.perceptronModelFile);
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+            String line;
+            String currentDocument = "";
+            System.out.println("STARTED -- Now reading input...");
+            while((line = reader.readLine()) != null) {
+                if (line.length() == 0) {
```

Comment:
could this block be in a separate function? or maybe move it to after the while loop?

As far as i understood, this only runs once, when you read the last line. So it should be clearer to have it outside the while loop.

Reply:
When there's an empty line, prediction happens and the code continues to read input afterwards. So basically, multiple inputs are separated by an empty line, and this can be run multiple times.